### PR TITLE
[PATCH v2] validation: ipsec: fix reassembly vector population

### PR DESCRIPTION
**Pull request by asasidharan**
Move the test vectors to source file and declare them as extern in
header file to avoid the creation of multiple instances of test vectors

Signed-off-by: Vidya Sagar Velumuri <vvelumuri@marvell.com>
Signed-off-by: Aakash Sasidharan <asasidharan@marvell.com>
Reviewed-by: Anoob Joseph <anoobj@marvell.com>

### CODE REVIEW

File: test/validation/api/ipsec/reass_test_vectors.c [hunk 1]
```diff
@@ -0,0 +1,354 @@
+ /* Copyright (c) 2021, Marvell
```
Comment (JannePeltonen) [inline test/validation/api/ipsec/reass_test_vectors.c:1]: There is an extra space in the beginning of the line.

File: test/validation/api/ipsec/reass_test_vectors.c [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier:     BSD-3-Clause
+ */
+
+#include "ipsec.h"
+#include <odp_api.h>
```
Comment (JannePeltonen) [inline test/validation/api/ipsec/reass_test_vectors.c:8]: odp_api.h does not need to be included here.

File: test/validation/api/ipsec/reass_test_vectors.h [hunk 3]
```diff
@@ -391,5 +63,4 @@ reass_test_vectors_init(void)
 	test_vector_payload_populate(&pkt_ipv4_udp_p2_f3, false);
 	test_vector_payload_populate(&pkt_ipv4_udp_p2_f4, false);
 }
-
```
Comment (JannePeltonen) [inline test/validation/api/ipsec/reass_test_vectors.h:394]: This white space removal is unnecessary (maybe accidental).

**Comment by JannePeltonen** [comment]
This looks ok apart from the couple of very minor cosmetic issues. Please add my reviewed-by tag in the next version.

File: test/validation/api/ipsec/reass_test_vectors.c (hunk 1 above)
Comment (asasidharan) [reply test/validation/api/ipsec/reass_test_vectors.c]: Ack.

File: test/validation/api/ipsec/reass_test_vectors.c (hunk 2 above)
Comment (asasidharan) [reply test/validation/api/ipsec/reass_test_vectors.c]: Ack.

File: test/validation/api/ipsec/reass_test_vectors.h (hunk 3 above)
Comment (asasidharan) [reply test/validation/api/ipsec/reass_test_vectors.h]: Ack.

**Comment by asasidharan** [comment]
With v2, addressed the review comments and added Reviewed-by tag.

**Review by JannePeltonen — APPROVED** [review]

**Pull request merged**